File: java/autodesk/com/wpe/hiring/api/implementation/UserRecommendation.java
package autodesk.com.wpe.hiring.api.implementation;

public class UserRecommendation {

  String userId;
  String bookTitle;
  String isbnNumber;
  String authorName;
  String genreName;

  public String getUserId() {
    return userId;
  }

  public void setUserId(String userId) {
    this.userId = userId;
  }

  public String getBookTitle() {
    return bookTitle;
  }

  public void setBookTitle(String bookTitle) {
    this.bookTitle = bookTitle;
  }

  public String getIsbnNumber() {
    return isbnNumber;
  }

  public void setIsbnNumber(String isbnNumber) {
    this.isbnNumber = isbnNumber;
  }

  public String getAuthorName() {
    return authorName;
  }

  public void setAuthorName(String authorName) {
    this.authorName = authorName;
  }

  public String getGenreName() {
    return genreName;
  }

  public void setGenreName(String genreName) {
    this.genreName = genreName;
  }


}

File: node/api/implementation/UserRecommendation.js
class UserRecommendation {

  constructor() {
    this.userId;
    this.bookTitle;
    this.isbnNumber;
    this.authorName;
    this.genreName;
  }

  getUserId() {
    return this.userId;
  }

  setUserId(userId) {
    this.userId = userId;
  }

  getBookTitle() {
    return this.bookTitle;
  }

  setBookTitle(bookTitle) {
    this.bookTitle = bookTitle;
  }

  getIsbnNumber() {
    return this.isbnNumber;
  }

  setIsbnNumber(isbnNumber) {
    this.isbnNumber = isbnNumber;
  }

  getAuthorName() {
    return this.authorName;
  }

  setAuthorName(authorName) {
    this.authorName = authorName;
  }

  getGenreName() {
    return this.genreName;
  }

  setGenreName(genreName) {
    this.genreName = genreName;
  }

}

module.export = UserRecommendation;
